### ISSUE TITLE
Disable rename _0 _1 _2 etc .. in layers and styles files

### ISSUE THREAD

**Issue by andreaordonselli** [tomchadwin/qgis2web#1009]
qgis2web adds a progressive numbering equal to _0 _1 _2 etc .. in the name of the files referring to the layers and styles.
Can I know how to disable this rename keeping only the layer name?
Can you tell me which part of the qgis plugin code to modify?

Then I'll be careful not to have two layers with the same name in the QGIS project.
Thank you.

**Comment by boesiii** [comment]
What don't you like about the numbering?  Why do you want to change it?  There is no easy way to disable it unless you edit the python plugin code.  Depending on which mapping library you are using, OpenLayers or Leaflet, there are multiple places you will need to change.  There are many parts of the code that may use the layer name, such as the popup function, so I would use caution.

**Comment by andreaordonselli** [comment]
Hello and thanks for the reply,
I use qgis2web only with **openlayers** export to export different projects.
Some of these are static (layers change infrequently) while others are very dynamic (layers change approximately every 2 weeks).
After exporting, I usually edit the `layer.js` file adding the groups so as to show a structured legend like the following:
![image](https://user-images.githubusercontent.com/89784373/186404162-66d574d4-9c06-4bbf-8ff4-faafc2f087ac.png)

The deactivation of the progressive numbering _0 _1 _2 etc ... would allow me not to have to rewrite the structure of the groups every time and not to have to re-apply any changes inserted after `new ol.layer.Vector({` like `opacity`, `permalink`, and other code added.

I could simply take the `layer.js` already edited in the previous export, adding the code parts related to the new layers inserted in the map. What do you think?
Greetings.

**Comment by boesiii** [comment]
QGIS2WEB exports most layers to geojson format.  If only the layers' data changes why not export the layer from QGIS to geojson and simply add what is needed to that file and rename to .Js file?

**Comment by andreaordonselli** [comment]
This solution can work well when the layer data is changed. But in my case it is the number of layers inserted in the project that changes. Today I have a project with 3 layers to export, in 2 weeks they become 15, then they go back to 7.
So when I export again with QGIS2WEB it is difficult to apply changes made previously because the layer name changes every time based on the sort order.

To give an example, I edit the **layer.js** file adding the layer grouping (but I also add code parts after **new ol.layer.Vector**)
```
var layersList = [
new ol.layer.Group({
            title: '<b>Basemap</b>',
			openInLayerSwitcher: true,
            layers: [
lyr_OpenStreetMap_0,
]}),
new ol.layer.Group({
            title: '<b>WMS Catasto</b>',
            layers: [
lyr_WMSCatastoFogli_2,lyr_WMSCatastoStrade_3,lyr_WMSCatastoAcque_4,lyr_WMSCatastoParticellezoom200m_5,lyr_WMSCatastoFabbricatizoom200m_6,lyr_WMSCatastoVestizionizoom100m_7,
]}),
new ol.layer.Group({
            title: '<b>Vincoli e Tutele</b>',
            openInLayerSwitcher: true,
            layers: [
lyr_PaiEsondazioniagg2016_8,lyr_PaiFraneagg2016_9,lyr_ParchieRiserve_10,lyr_VincoloIdrogeologico_11,lyr_VincoliArcheologiciArt142m_12,lyr_GalassiniDM310785_13,lyr_BellezzeNaturaliArt136_14,lyr_ZPS_15,lyr_SIC_16,
]}),
new ol.layer.Group({
            title: '<b>Cartografia generale</b>',
            openInLayerSwitcher: true,
			
            layers: [
lyr_ToponimiMonti_17,lyr_ConfiniComunali_18,lyr_ConfiniProvinciali_19
]}),
];
```
So with each new export I find myself having to rewrite everything because the name of the layers always changes (before it was called **lyr_ParchieRiserve_10** but in the next project it is called **lyr_ParchieRiserve_4** etc ...)

If QGIS2WEB does not rename the layers, it would always be written **lyr_ParchieRiserve** and I would not have to rewrite the code but take copy the ones I modified in the previous export.

However, if, as you have already written to me, modifying the pyton code is complex, forget it. I tried this path not knowing if it was difficult.

Thank you for your support, let me know if you can think of a solution to my "problem".
Thank you have a nice day.

**Comment by boesiii** [comment]
I found all the areas of code that deal with the layer names.  There are numerous sections and files that add the underscore and number to the layer name.  I am not sure what each one does but I removed the underscore and number to each one until it worked.  You should replace your plugin files with the ones attached.   Reload the plugin then reload your project and export and see if it works.

[files_with_edits.zip](https://github.com/tomchadwin/qgis2web/files/9434045/files_with_edits.zip)

**Comment by andreaordonselli** [comment]
I have tested it and it works perfectly.
Really thank you very much.
It will greatly facilitate my work.

@tomchadwin  I recommend proceeding with this file version **injected into the official plugin**.

IMPROVE: 
Allows you to apply any changes made to the code more simply after new export

WORSE:
There cannot be 2 layers with the same name in the QGIS project, but this case never occurs otherwise the data in the map could not be recognized!

Thanks so much again.

**Comment by andreaordonselli** [comment]
I managed it by myself by modifying this single file you edited `olStyleScripts.py` I simply reintroduced in some parts the previous code that made up the png files with _0 _1 _2 etc ...
The result is perfect.
For simplicity I enclose the whole package with this last updated file.

[files_with_edits.zip](https://github.com/tomchadwin/qgis2web/files/9454404/files_with_edits.zip)

Thanks again, regards.
PS: it was the first time I was working on .py code so I asked for your help first
